controllers/configmap_controller.go

```diff
@@ -183,7 +184,13 @@ func (r *ConfigMapReconciler) Reconcile(ctx context.Context,
 	case servicescm.Name:
 		return ctrl.Result{}, r.reconcileServices(ctx, configMap)
 	case wiparser.InstanceConfigMap:
-		return ctrl.Result{}, r.reconcileNodes(ctx, configMap)
+		err := r.reconcileNodes(ctx, configMap)
+		var upgradeErr *UpgradeLimitExceededError
+		if errors.As(err, &upgradeErr) {
+			r.log.Info(upgradeErr.Error())
+		} else {
+			return ctrl.Result{}, err
+		}
```

Comment:
I recommend to invert the execution flow in a positive way, taking the corner cases as branches with conditional statements. for example

```go

// if error is unknown, throw it
if !errors.As(err, &upgradeErr) {
  return ctrl.Result{}, err
}

// log info and retry with `.Result{Requeue: true}`

```

Reply:
Doing it this way feels a little weird to me, because  it makes it feel like logging and retrying is the default behavior, instead of the upgrade limit exceeded error being the corner case

Reply:
I agree and as suggested earlier on slack consider returning {Requeue: true}